### PR TITLE
SDL: Always set X, Y, Shift and Control in mouse input events

### PR DESCRIPTION
**Pull request by Desour**
These fields are always valid (see `IEventReceiver.h`).
And the minetest formspecs need them for item interaction.

Test instructions:
Open inv and move items. Try holding shift. Also try mouse wheel.

### CODE REVIEW

**Comment by Desour** [comment]
Rebased (not because needed, but to retry cancelled macos build).

File: source/Irrlicht/CIrrDeviceSDL.cpp [hunk 1]
```diff
@@ -623,6 +632,10 @@ bool CIrrDeviceSDL::run()
 				irrevent.KeyInput.Char = 0;
 				if (mp.SDLKey == SDLK_DELETE || mp.SDLKey == SDLK_RETURN || mp.SDLKey == SDLK_BACKSPACE || irrevent.KeyInput.Control)
 					irrevent.KeyInput.Char = mp.SDLKey;
+				if (mp.SDLKey == SDLK_LSHIFT || mp.SDLKey == SDLK_RSHIFT)
+					IsShiftDown = (SDL_event.type == SDL_KEYDOWN);
+				if (mp.SDLKey == SDLK_LCTRL || mp.SDLKey == SDLK_RCTRL)
+					IsControlDown = (SDL_event.type == SDL_KEYDOWN);
```
Comment (sfan5) [inline source/Irrlicht/CIrrDeviceSDL.cpp:638]: it really has no better way for this?

Comment (Desour) [reply source/Irrlicht/CIrrDeviceSDL.cpp:638]: There's `SDL_GetKeyboardState` to get all keys and `SDL_GetModState` to get modifiers (i.e. shift and control). I guess the latter could be used.
Will do.

Comment (Desour) [reply source/Irrlicht/CIrrDeviceSDL.cpp]: Done. And rebased.

**Review by sfan5 — APPROVED** [review]
lgtm

**Pull request merged**